lib/chef/azure/commands/enable.rb

```diff
@@ -178,6 +179,23 @@ def configure_chef_only_once
     end
   end
 
+  def load_cloud_attributes_in_hints
+    cloud_attributes = {}
+    cloud_attributes["vm_name"] = Socket.gethostname
+    if windows?
+      vm_dns =  shell_out!("ipconfig").stdout
+      cloud_attributes["fqdn"] = vm_dns.gsub(/[a-zA-Z0-9-]*.[a-zA-Z0-9]*.[a-zA-Z0-9]*.cloudapp.net/).first.split('.')[0] + '.cloudapp.net' if vm_dns

```

Comment:
I guess my question is: to do this, did you have to figure out how Azure dns names work and encode this yourself, or did this approach come from some other code? It mostly matters because they may change their DNS naming scheme at some point, which means we'll be broken.